src/coreclr/jit/importer.cpp

```diff
@@ -3724,6 +3729,15 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
     }
 #endif // FEATURE_HW_INTRINSICS
 
+    if (!isIntrinsic)
+    {
+        // TODO-Cleanup: Outside the cases above, there are many intrinsics which
```

Comment:
What would we want to cleanup here? It looks good as is to me. Nothing to cleanup.

Reply:
I had seen some chatter about people liking the fact that `AltJit` doesn't match `corelib` only and so it would make checking isolated code samples (such as those using `IsKnownConstant`) simpler.

This check prevents such scenarios from working which may be undesirable.

Reply:
I've removed the `TODO-Cleanup` and left this just as a comment explaining why we exit early.